plugin_guide.md

```diff
@@ -0,0 +1,184 @@
+
```

Comment:
Need to see how this aligns with https://ewatercycle.readthedocs.io/en/latest/adding_models.html
Perhaps it can fully replace it at some point